python/packages/ai/teams/ai/models/openai_model.py

```diff
@@ -124,11 +125,47 @@ async def complete_prompt(
         template: PromptTemplate,
     ) -> PromptResponse[str]:
         max_tokens = template.config.completion.max_input_tokens
+
+        # Setup actions if enabled
+        is_tools_aug = (
+            template.config.augmentation
+            and template.config.augmentation.augmentation_type == "tools"
+        )
+        tool_choice = (
+            template.config.completion.tool_choice
+            if template.config.completion.tool_choice is not None
+            else "auto"
+        )
+        parallel_tool_calls = (
+            template.config.completion.parallel_tool_calls
+            if template.config.completion.parallel_tool_calls is not None
+            else True
+        )
+        tools: List[chat.ChatCompletionToolParam] = []
+
+        # If tools is enabled, reformat actions to schema
+        if is_tools_aug:
+            if not template.actions or len(template.actions) == 0:
```

Comment:
this should not throw an exception, you can use tools while not providing any tools without the api throwing an error

Reply:
what would be the purpose of using tools aug without specifying any tools in `actions.json`? or do you mean the onus is on the developer to configure correctly